skflow/trainer.py

```diff
@@ -45,20 +45,35 @@ class TensorFlowTrainer(object):
       gradients: Gradients tensor.
     """
 
-    def __init__(self, loss, global_step, optimizer, learning_rate, clip_gradients=5.0):
+    def __init__(self, loss, global_step, optimizer,
+                 learning_rate, clip_gradients=5.0):
         """Build a trainer part of graph.
 
         Args:
           loss: Tensor that evaluates to model's loss.
           global_step: Tensor with global step of the model.
           optimizer: Name of the optimizer class (SGD, Adam, Adagrad) or class.
+          learning_rate: If this is constant float value, no decay function is used.
+                         Instead, a customized decay function can be passed that accepts
+                         global_step as parameter and returns a Tensor.
+                         e.g. exponential decay function:
+                         def exp_decay(global_step):
+                            return tf.train.exponential_decay(
+                                learning_rate=0.1, global_step=global_step,
+                                decay_steps=2, decay_rate=0.001)
         """
         self.loss = loss
         self.global_step = global_step
-        self._learning_rate = tf.get_variable(
-            "learning_rate",
-            [],
-            initializer=tf.constant_initializer(learning_rate))
+        # pylint: disable=redefined-variable-type
+        if isinstance(learning_rate, float):
+            self._learning_rate = tf.get_variable(
+                "learning_rate",
+                [],
+                initializer=tf.constant_initializer(learning_rate))
+        else:
```

Comment:
this may a bit easier to read in next way:
if isinstance(learning_rate, float):
   ...
elif callable(learning_rate):
  ...
else:
  raise ValueError(...)

Also you should add Raises: section into the __init__ doc string.